plugins/discourse-policy/assets/javascripts/discourse/initializers/rich-editor-extension.js

```diff
@@ -0,0 +1,160 @@
+import { camelize } from "@ember/string";
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+const POLICY_ATTRIBUTES = {
+  group: { default: null },
+  groups: { default: null },
+  version: { default: "1" },
+  accept: { default: null },
+  revoke: { default: null },
+  reminder: { default: null },
+  renewStart: { default: null },
+  private: { default: null },
+};
+
+/** @type {RichEditorExtension} */
+const extension = {
+  nodeSpec: {
+    policy: {
+      attrs: POLICY_ATTRIBUTES,
+      group: "block",
+      content: "block*",
+      createGapCursor: true,
+      parseDOM: [
+        {
+          tag: "div.policy",
+          getAttrs: (dom) => {
+            const attrs = {};
+            // Convert data-* attributes to camelCase node attributes
+            Object.entries(dom.dataset).forEach(([key, value]) => {
+              if (value && value.trim() !== "" && value !== '""') {
+                attrs[camelize(key)] = value;
+              }
+            });
+            return attrs;
+          },
+        },
+      ],
+      toDOM(node) {
+        const attrs = { class: "policy" };
+
+        // Set data attributes, converting camelCase back to kebab-case and filtering out empty values
+        Object.entries(node.attrs).forEach(([key, value]) => {
+          if (value && value.trim() !== "" && value !== '""') {
+            const dataKey = key.replace(/([A-Z])/g, "-$1").toLowerCase();
+            attrs[`data-${dataKey}`] = value;
+          }
+        });
```

Comment:
We seem to be doing this case conversion a few times, might be worth extracting into an helper? (likewise for the reverse operation)